### PR TITLE
finish ticket 8 - add comment

### PR DESCRIPTION
**Pull request by charlieBdev**

### CODE REVIEW

File: src/components/CommentForm.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (newComment.length <= 50) {
+            postComment(article_id, user, newComment)
+            .then(() => {
+                setIsAdded(true)
+            })
+            .catch(() => {
+                setIsAdded(false)
+            })
+            setComments((currComments) => {
```
Comment (itskatherine) [inline src/components/CommentForm.jsx]: You're currently getting an error caused by new comments not having keys 🔑 
You use `comment_id` for your key, which is a good move as it's a unique value
However it's `undefined` for new comments, as the `comment_id` is only established once the comment has been added to the database.
You would however be able to access the comment id of the new added comment in what is returned from the patch API request. Can you please implement this so when this comment is set, it also includes the unique comment id.

The comment state is also being updated when the patch request doesnt go through (like when there is no internet), which gives a false impression to the user that their comment has been posted successfully. Can you change this so the state is only updated if the patch request is successful?

File: src/components/CommentForm.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return (
+        <form onSubmit={handleSumbit}>
+            <textarea rows="3" cols="25" type="text" placeholder="Add comment" value={newComment} onChange={(event) => setNewComment(event.target.value)}/>
+            {isError ? (<p>Your message is too long!</p>) : (<p>{50 - newComment.length} chars left</p>)}
+            <button disabled={newComment.length > 50 || newComment.length < 1} type="submit" value="submit">Comment</button>
+            {isAdded && <p>Comment added!</p>}
+        </form>
+    )
+}
+
```
Comment (itskatherine) [inline src/components/CommentForm.jsx:41]: Nice checking for valid input 🙌

**Review by itskatherine — COMMENTED** [review]
Hi Charlie, a few things to look at here but solid so far
Could you please resubmit this PR once you've addressed my comments 👍

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Charlie, almost there! I'm wondering why the character limit is only 50, that would allow only about 7 or 8 words as a comment.

The 'Comment' button should be disabled for the time of posting to avoid posting the same comment multiple times. Please fix that. Once it's done, it's good to merge.

**Pull request merged**